tools/tools.go

```diff
@@ -0,0 +1,27 @@
+// +build tools
+
+// This file ensures tool dependencies are kept in sync. This is the
+// recommended way of doing this according to
+// https://github.com/golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module
+
+// To install the following tools at the version used by this repo run:
+// $ make tools
+// or
+// $ go generate -tags tools tools/tools.go
+
+package tools
+
+//go:generate go install github.com/golang/protobuf/proto
+import _ "github.com/golang/protobuf/proto"
+
+//go:generate go install github.com/gogo/protobuf/gogoproto
+import _ "github.com/gogo/protobuf/gogoproto"
+
+//go:generate go install github.com/gogo/protobuf/protoc-gen-gogo
+import _ "github.com/gogo/protobuf/protoc-gen-gogo"
+
+//go:generate go install github.com/gogo/protobuf/protoc-gen-gofast
+import _ "github.com/gogo/protobuf/protoc-gen-gofast"
+
+//go:generate go install github.com/golangci/golangci-lint/cmd/golangci-lint
+import _ "github.com/golangci/golangci-lint/cmd/golangci-lint"
```

Comment:
Even if the module is not imported, it is confirmed that the module is installed by `go install`. Can you explain the reason for the import modules?

Reply:
Thanks for the comment. Sorry I didn't fully understand the question. But the behavior I understood is below.

Typing `make tools` will run `go generate -tags tools tools/tools.go`. `go generate` runs comments in patterns that begin with `go:generate`. And modules imported into `tools.go` are not used directly in the source code, but are tracked by go mod. For more details please refer below link:

https://github.com/golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module

Reply:
Thank you for your feedback! 👍